contracts/cw20-merkle-airdrop/src/contract.rs

```diff
@@ -212,6 +213,56 @@ pub fn execute_claim(
     Ok(res)
 }
 
+pub fn execute_burn(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    stage: u8,
+) -> Result<Response, ContractError> {
+    // authorize owner
+    let cfg = CONFIG.load(deps.storage)?;
+    let owner = cfg.owner.ok_or(ContractError::Unauthorized {})?;
+    if info.sender != owner {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // make sure is expired
+    let expiration = STAGE_EXPIRATION.load(deps.storage, stage)?;
+    if !expiration.is_expired(&env.block) {
+        return Err(ContractError::StageNotExpired { stage, expiration });
+    }
+
+    // Get balance
+    let balance: BalanceResponse = deps
```

Comment:
If multiple stages are going on at the same time, this code will result in wiping out of full airdrop balance 💣 Best to save stage airdrop amount on register_merkle_root phase and burn that amount

Reply:
I had to add also a "total claimed per stage" that updates every time an user claim, but should be ok now